### ISSUE TITLE
Missing vmConfig Parameters in kubernetes/VirtualMachineInfraConfigPolicies policy

### ISSUE THREAD

**Issue by sandkum5** [CiscoDevNet/intersight-powershell#57]
**The parameters Cluster, Datastore, and Passphrase are missing under VmConfig Object in the PowerShell module: Initialize-IntersightKubernetesBaseVirtualMachineInfraConfig**

```
> Get-Help New-IntersightKubernetesVirtualMachineInfraConfigPolicy -detailed
    -VmConfig <KubernetesBaseVirtualMachineInfraConfig>
        Virtual machine infrastucture provider allocation properties.
        
        Note :- Use Initialize-IntersightKubernetesBaseVirtualMachineInfraConfig to create the object of complex type 
        KubernetesBaseVirtualMachineInfraConfig
```
```
> Get-help Initialize-IntersightKubernetesBaseVirtualMachineInfraConfig -detailed
SYNTAX
    Initialize-IntersightKubernetesBaseVirtualMachineInfraConfig [-AdditionalProperties 
    <System.Collections.Generic.Dictionary`2[string,object]>] [-ClassId 
    <KubernetesBaseVirtualMachineInfraConfig.ClassIdEnum>] [-Interfaces <System.Collections.Generic.List`1[string]>] 
    [-NetworkInterfaces <System.Collections.Generic.List`1[Model.KubernetesNetworkInterfaceSpec]>] [-ObjectType 
    <KubernetesBaseVirtualMachineInfraConfig.ObjectTypeEnum>] [-Json <SwitchParameter>] [-WithHttpInfo <SwitchParameter>] 
    [<CommonParameters>]
```

The parameters are missing from the Intersight Browser REST API Client Request Model as well. Probably that's why they are missing from PowerShell Module.  

A post call to create a VM Infra Config Policy works fine in Intersight.
Works fine:
```
POST https://intersight.com/api/v1/kubernetes/VirtualMachineInfraConfigPolicies
{
    "Name":"demo-2",
    "Organization":{"ObjectType":"organization.Organization","Moid":"xxxxxxxxxx"},
    "Target":"xxxxx",
    "VmConfig":{
        "ObjectType":"kubernetes.EsxiVirtualMachineInfraConfig",
         "Cluster":"xxxxx",
        "Datastore":"xxxxx",
        "Interfaces":["xxxx"],
        "Passphrase":"xxxxxxx"
        }
}
```

**Comment by Ghufz** [comment]
use the cmdlet  ```powershell Initialize-IntersightkubernetesEsxiVirtualMachineInfraConfig ``` to do the above configuration.

If you use the cmdlet Initialize-IntersightKubernetesBaseVirtualMachineInfraConfig then pass the properties which are not part of parameter as AdditionalProperties. 

Thanks for your feedback we will update the help content for the same.

**Comment by sandkum5** [comment]
I tried using the new module but it doesn't set the passphrase in the created policy. Everything else looks good. 

> $vmConfigObject = Initialize-IntersightKubernetesEsxiVirtualMachineInfraConfig -Cluster $cluster -Datastore $datastore -Interfaces $interfaces -Passphrase $passphrase

When I am looking at the created object, the flag IsPassphraseSet is set to False. I don't see any option to add this flag. 

> $vmConfigObject
ClassId              : KubernetesEsxiVirtualMachineInfraConfig
ObjectType           : KubernetesEsxiVirtualMachineInfraConfig
Cluster              : cx-hxe1
Datastore            : SpringpathDS-WZP2325004C
IsPassphraseSet      : False
Passphrase           : adfadfasdaljdlfjds
ResourcePool         : 
AdditionalProperties : {}
Interfaces           : {Management Network}
NetworkInterfaces    : 

I tried passing the flag under additional properties but it failed with an error. 

$vmConfigObject = Initialize-IntersightKubernetesEsxiVirtualMachineInfraConfig -Cluster $cluster -Datastore $datastore -Interfaces $interfaces -Passphrase $passphrase -ClassId "KubernetesEsxiVirtualMachineInfraConfig" -ObjectType "KubernetesEsxiVirtualMachineInfraConfig" -AdditionalProperties @{ IsPassphraseSet = True }

Response: 
Initialize-IntersightKubernetesEsxiVirtualMachineInfraConfig: Cannot bind parameter 'AdditionalProperties'. Cannot create object of type "System.Collections.Generic.Dictionary`2[System.String,System.Object]". The IsPassphraseSet property was not found for the System.Collections.Generic.Dictionary`2[[System.String, System.Private.CoreLib, Version=[6.0.0.0](https://6.0.0.0/), Culture=neutral, PublicKeyToken=7cec85d7bea7798e

Kindly advise.
